### ISSUE TITLE
Style/ArgumentsForwarding is detecting false positives

### ISSUE THREAD

**Issue by dylan-hoefsloot** [rubocop/rubocop#12621]
I am getting false positives, which if autocorrected break the code

Here is an example:
```ruby
def before_transition(options = {}, &block)
  add_callback(callback_type: :before, callback_class: Callback, from: options[:from], to: options[:to], &block)
end
```

when I run `rubocop` I get the warning:
```
Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
```

Which seemed incorrect to me, but just in case I was missing something I tried running autocorrect which produced:
```ruby
def before_transition(...)
  add_callback(...)
end
```

Which is completely broken and a change of behaviour.

--------

## Expected behavior

No warning raised

## Actual behavior

Raised a false positive

## Steps to reproduce the problem

As above

## RuboCop version

```
$ [bundle exec] rubocop -V
1.60.0 (using Parser 3.3.0.4, rubocop-ast 1.30.0, running on ruby 3.2.1) [arm64-darwin22]
  - rubocop-performance 1.20.2
  - rubocop-rspec 2.26.1
```

**Comment by AlexWayfer** [comment]
I have kind of similar problem:

```ruby
def view(path = nil, options = {}, &block)
  cache = options.delete(:cache)
  cache = config[:environment] == 'production' if cache.nil?
  template = Flame::Render.new(
    self,
    (path || caller_locations(1, 1)[0].label.to_sym),
    options
  )
  template.render(cache: cache, &block)
end
```

```
lib/flame/controller.rb:128:12: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
		def view(path = nil, options = {}, &block)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/flame/controller.rb:136:20: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
			template.render(cache: cache, &block)
                   ^^^^^^^^^^^^^^^^^^^^
```

![image](https://github.com/rubocop/rubocop/assets/6510020/5f1afad9-9cc6-477c-81bf-6d05c69198f9)

How should I replace there something via `...`?

RuboCop v1.60, Target Ruby version is 3.0.

**Comment by koic** [comment]
The issue is caused by the same reason as #12618 and is planned to be resolved in #12624.
